Code/src/pipeline/catboost_pipeline.py

```diff
@@ -0,0 +1,63 @@
+from pipeline import ModelPipeline
+
+class CatBoostPipeline(ModelPipeline):
```

Comment:
```suggestion
class CatBoostPipeline(BaseModelPipeline):
```

Not a strict rule but with the intent that `ModelPipeline` won't be used by folks, only the `CatBoostPipeline` and `GNNPipeline`, renaming `ModelPipeline` to `BaseModelPipeline` can make that more clear. This suggestion would require renaming `ModelPipeline` to `BaseModelPipeline` elsewhere.